src/components/ModalWithForm.js

```diff
@@ -2,13 +2,26 @@ import Modal from "./Modal.js";
 
 export default class ModalWithForm extends Modal {
   //---------------MODALWITHFORM CONSTRUCTOR------------------------>>
-  constructor(modalSelector, formSubmit, { formSelector }) {
+  constructor(
+    modalSelector,
+    formSubmit,
+    { formSelector, submitButtonSelector }
+  ) {
     super(modalSelector);
     this._form = this._modalElement.querySelector(formSelector);
     this._formSubmit = formSubmit;
+    this._button = this._modalElement.querySelector(submitButtonSelector);
+    this._originalButtonText = this._button.textContent;
     this._inputList = Array.from(this._form.querySelectorAll(".modal__input"));
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`